kroxylicious-systemtests/src/test/java/io/kroxylicious/systemtests/OperatorChangeDetectionST.java

```diff
@@ -287,6 +317,24 @@ private static void assertDeploymentUpdated(String namespace, String originalChe
         LOGGER.info("New checksum: {}", newChecksumFromAnnotation);
     }
 
+    private static void assertDeploymentUnchanged(String namespace, String originalChecksum, int expectedReplicaCount) {
```

Comment:
does ` resourceManager.createOrUpdateResourceWithWait` wait for the KafkaProxy status to be updated? Just worried about the risks of an `unchanged` assertion where we check moments before the change.

Reply:
I think it does but the assertion checks that the deployment has the expected replica count so it doesn't just see an unchanged checksum and return.